apps/backend/src/modules/users/user.repository.ts

```diff
@@ -47,10 +49,22 @@ class UserRepository implements Repository {
 			.query()
 			.findById(id)
 			.whereNull("deletedAt")
-			.returning("*")
-			.execute();
+			.withGraphFetched("groups.permissions")
+			.modifyGraph("groups", (builder) => {
+				builder.select("name");
+			})
+			.modifyGraph("groups.permissions", (builder) => {
+				builder.select("name", "key");
+			})
+			.castTo<
+				{
+					deletedAt: string;
+					passwordHash: string;
+					passwordSalt: string;
+				} & UserAuthResponseDto
+			>();
 
-		return user ? UserEntity.initialize(user) : null;
+		return UserEntity.initialize(user);
```

Comment:
what if we didn't find a user

Reply:
I wanted to leave it, but the linter gave the error `Unnecessary conditional, value is always truthful.`

Reply:
we can't leave it as it is now, because this will cause unexpected application error

Reply:
The problem is that you cast the return query type to be found user so linter thinks it is always be found